internal/controller/pkg/resolver/reconciler.go

```diff
@@ -172,9 +173,18 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{}, errors.Wrap(resource.IgnoreNotFound(err), errGetLock)
 	}
 
-	// NOTE(hasheddan): we add a finalizer and never remove it to guard against
-	// a user accidentally deleting the lock, which should never be done.
-	// Consider instead recreating the lock if it is not present.
+	// If no packages exist in Lock then we remove finalizer and wait until a
+	// package is added to reconcile again. This allows for cleanup of the Lock
+	// when uninstalling Crossplane after all packages have already been
+	// uninstalled.
+	if len(lock.Packages) == 0 {
+		if err := r.lock.RemoveFinalizer(ctx, lock); err != nil {
+			log.Debug(errRemoveFinalizer, "error", err)
+			return reconcile.Result{RequeueAfter: shortWait}, nil
+		}
+		return reconcile.Result{}, nil
```

Comment:
Wouldn't this make that object never reconcile because the array has no element when the object is created?

Reply:
Oh nevermind, I always think that the list is constructed by this controller but it's not, so it will come as filled by other controllers.